src/generate.rs

```diff
@@ -461,69 +460,135 @@ pub fn peripheral(
+                    .iter()
+                    .all(|element| element.parse::<usize>().is_ok());
+                
+                let sequential_indexes = if numeral_indexes && sequential_adresses {
+                    array_info.dim_index.clone()
+                        .unwrap()
+                        .iter()
+                        .map(|element| element.parse::<u32>().unwrap())
+                        .collect::<Vec<u32>>()
+                        .eq(&(0..array_info.dim).collect::<Vec<u32>>())
+                } else {
+                    false
+                };
+
+                let array_convertible = sequential_indexes && numeral_indexes && sequential_adresses;
+
+                if array_convertible {
+                    registers_expanded.push(
+                        RegisterBlockField{
+                            field: util::convert_svd_register(&register),
+                            description: Ident::from(info.description.clone()),
+                            offset: info.address_offset,
+                            size: register_size * array_info.dim,
+                        });                                    
+                } else {
+                    let mut field_num = 0;
+                    for field in util::expand_svd_register(register).iter() {
+                        registers_expanded.push(
+                            RegisterBlockField{
+                                field: field.clone(),
+                                description: Ident::from(info.description.clone()),
+                                offset: info.address_offset + field_num * array_info.dim_increment,
+                                size: register_size,
+                            });
+                        field_num += 1;
+                    }
+                }
+            },
+        }
+    }
+
+    registers_expanded.sort_by_key(|x| x.offset);
+
+    for register in registers_expanded {
         let pad = if let Some(pad) = register.offset.checked_sub(offset) {
             pad
         } else {
             writeln!(
                 io::stderr(),
                 "WARNING {} overlaps with another register at offset {}. \
                  Ignoring.",
-                register.name,
+                register.field.ident.unwrap(),
                 register.offset
-            ).ok();
+            )
+                .ok();
             continue;
         };
 
         if pad != 0 {
             let name = Ident::new(format!("_reserved{}", i));
             let pad = pad as usize;
-            fields.push(quote! {
-                #name : [u8; #pad],
-            });
+            fields.append(
+                quote! {
+                    #name : [u8; #pad],
+                });
             i += 1;
         }
 
         let comment = &format!(
             "0x{:02x} - {}",
             register.offset,
-            util::respace(&register.info.description)
+            register.description,
```

Comment:
Please revert this change or the documentation will have random newlines in it. As in: keep the `util::respace`.